src/apps/bridge/sensor_drivers/rbrCodaSensor.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include "FreeRTOS.h"
+#include "abstractSensor.h"
+#include "avgSampler.h"
+#include "bm_rbr_data_msg.h"
+#include "sensorController.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#define RBR_CODA_NUM_SAMPLE_MEMBERS 3
+
+typedef struct rbr_coda_aggregations_s {
+  double temp_mean_deg_c;
+  double pressure_mean_ubar;
+  double pressure_stdev_ubar;
+  uint32_t reading_count;
+  BmRbrDataMsg::SensorType_t sensor_type;
+} rbr_coda_aggregations_t;
+
+typedef struct RbrCodaSensor : public AbstractSensor {
+  uint32_t rbr_coda_agg_period_ms;
+  AveragingSampler temp_deg_c;
+  AveragingSampler pressure_ubar;
+  uint32_t reading_count;
+  BmRbrDataMsg::SensorType_t latest_sensor_type;
+
+  // Extra sample padding to account for timing slop. Calculated as the sample frequency + 2 minutes bridge on period + some extra slop.
+  // 2 minutes is the minimum bridge on period and the soft by default is sampling at 2Hz. So 2*120 + 30 = 270.
+  static constexpr uint32_t N_SAMPLES_PAD = 270;
+  static constexpr uint8_t MIN_READINGS_FOR_AGGREGATION = 3;
+  static constexpr double TEMP_SAMPLE_MEMBER_MIN = -20;
+  static constexpr double TEMP_SAMPLE_MEMBER_MAX = 40;
+  static constexpr double PRESSURE_SAMPLE_MEMBER_MIN = 0;
+  static constexpr double PRESSURE_SAMPLE_MEMBER_MAX = 20970000;
+  static constexpr double PRESSURE_STDEV_SAMPLE_MEMBER_MIN = 0;
+  static constexpr double PRESSURE_STDEV_SAMPLE_MEMBER_MAX = 327000;
```

Comment:
these are based on the BMSensorDataMessage1 sheet.

Reply:
You can use the values in dbar here:
https://www.notion.so/sofarocean/SM-product-spec-RBR-Coda-Adapter-Module-bcfcdfa5f3fa4ca2811415089d2f8a29?pvs=4#541374f4275f4b8998d1d7fd6eab15c4

Reply:
sounds good!